src/cdk/tree/tree.spec.ts

```diff
@@ -35,6 +35,20 @@ describe('CdkTree', () => {
     }).compileComponents();
   }
 
+  it('should clear out the `mostRecentTreeNode` on destroy', () => {
+    configureCdkTreeTestingModule([SimpleCdkTreeApp]);
+    const fixture = TestBed.createComponent(SimpleCdkTreeApp);
+    fixture.detectChanges();
+
+    // Cast the assertions to a boolean to avoid Jasmine going into an
+    // infinite loop when stringifying the object, if the test starts failing.
+    expect(!!CdkTreeNode.mostRecentTreeNode).toBe(true);
+
+    fixture.destroy();
+
+    expect(!!CdkTreeNode.mostRecentTreeNode).toBe(false);
```

Comment:
I left a comment about it a few lines about. Using `.toBeFalsy` will cause Jasmine to go into an infinite loop, if this test ever starts failing, because it tries to stringify an object with circular references.